src/test/java/com/aws/greengrass/logmanager/model/LogFileGroupTest.java

```diff
@@ -0,0 +1,52 @@
+package com.aws.greengrass.logmanager.model;
+
+import com.aws.greengrass.logmanager.exceptions.InvalidLogGroupException;
+import com.aws.greengrass.testcommons.testutilities.GGExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+import static com.aws.greengrass.logmanager.LogManagerService.ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG;
+import static com.aws.greengrass.logmanager.util.UnitTestLogFileHelper.givenAStringOfSize;
+import static com.aws.greengrass.logmanager.util.UnitTestLogFileHelper.writeFile;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith({MockitoExtension.class, GGExtension.class})
+public class LogFileGroupTest {
+    @TempDir
+    static Path directoryPath;
+
+    @Test
+    void GIVEN_log_files_THEN_find_the_active_file() throws IOException, InterruptedException,
+            InvalidLogGroupException {
+        ACTIVE_LOG_FILE_FEATURE_ENABLED_FLAG.set(true);
+        LogFile file = new LogFile(directoryPath.resolve("greengrass_test.log_1").toUri());
+        byte[] bytesArray = givenAStringOfSize(1024).getBytes(StandardCharsets.UTF_8);
+        writeFile(file, bytesArray);
+
+        TimeUnit.SECONDS.sleep(1);
```

Comment:
Why this sleep?

Reply:
This shouldn't be needed. Nothing here is running in different threads.

Reply:
This is only for lazily differ the creation time of two files, then we know the last file is the active file.

Reply:
Add a comment that says why we're doing this then.